File: LogicalThinkingPatterns/InvertedPyramid/InvertedPyramid.py
class InvertedPyramid:
    def printPattern(self, n):
        for i in range(n, 0, -1):
            pattern = ""
            for j in range(n-i, 0, -1):
                pattern += " "
            for k in range(0, i):
                pattern += "* "
            print(pattern)
    
    def run(self):
        n = int(input("Enter the number of rows: "))
        self.printPattern(n)


starInverted = InvertedPyramid()
starInverted.run()

File: LogicalThinkingPatterns/InvertedPyramid/InvertedPyramid.ts
class InvertedPyramid {
  printPattern(n: number): void {
    for (let i = n; i > 0; i--) {
      let pattern: string = "";
      for (let j = n - i; j > 0; j--) {
        pattern += " ";
      }
      for (let k = 0; k < i; k++) {
        pattern += "* ";
      }
      console.log(pattern);
    }
  }
}

const starInverted = new InvertedPyramid();
starInverted.printPattern(5);
